implants/lib/eldritch/src/file/find_impl.rs

```diff
@@ -0,0 +1,80 @@
+use std::{path::{Path, PathBuf}, time::{SystemTime, UNIX_EPOCH}};
+use anyhow::{anyhow, Result};
+#[cfg(unix)]
+use std::os::unix::fs::PermissionsExt;
+
+fn check_path(path: &PathBuf,  name: Option<String>, file_type: Option<String>, permissions: Option<u64>, modified_time: Option<u64>, create_time: Option<u64>) -> Result<bool> {
+    if let Some(name) = name {
+        if !path.file_name().unwrap().to_str().unwrap().contains(&name) {
+            return Ok(false);
+        }
+    }
+    if let Some(file_type) = file_type {
+        if !path.is_file() && file_type == "file" {
+            return Ok(false);
+        }
+        if !path.is_dir() && file_type == "dir" {
+            return Ok(false);
+        }
+    }
+    if let Some(permissions) = permissions {
+        let metadata = path.metadata()?.permissions();
+        #[cfg(unix)]
+        {
+            if metadata.mode() != (permissions as u32) {
+                return Ok(false);
+            }
+        }
+        #[cfg(windows)]
+        {
+            if permissions == 0 && metadata.readonly() {
+                return Ok(false);
+            }
+            if permissions == 1 && !metadata.readonly() {
+                return Ok(false);
+            }
+        }
+    }
+    if let Some(modified_time) = modified_time {
```

Comment:
Outside the scope of this PR but in the future is it possible to add newer than or older than?

Seeing this I realized it'll be be rare files match a specific second.

Reply:
Yeah that was what I was curious about with these. Do we want older/newer to be the default action?

Reply:
If you want to make the changes in this PR i would:
- Remove the modified time and created time fields
- Replace modified time with a modified_before and modified_after option
- Replace created time with a created_before and created_after option

This would make it possible to time bound your search for things created within a specifir and or modified in a specific range.